addon/helpers/hook.js

```diff
@@ -6,10 +6,38 @@ import returnWhenTesting from 'ember-hook/utils/return-when-testing';
 
 const { Helper } = Ember;
 
-export default Helper.extend({
-  compute(params, qualifiers = {}) {
-    const [hook] = params;
+/**
+ * Test if an object is empty `Ember.isEmpty` surprisingly doesn't do that
+ * @see {@link https://github.com/emberjs/ember.js/issues/5543}
+ * @see {@link https://github.com/emberjs/ember.js/blob/v2.10.0/packages/ember-metal/lib/is_empty.js}
+ *
+ * @param {Object} obj - the object to test
+ * @returns {Boolean} true if there are no keys on the object
+ */
+function isEmpty (obj) {
+  // Ember.keys is being deprecated in favor of Object.keys, handle both cases
+  if (Ember.keys) {
+    return Ember.keys(obj).length === 0;
+  } else if (Object.keys) {
+    return Object.keys(obj).length === 0;
+  }
+
+  // if we can't tell if it's empty, just assume it's empty, so the default case is used
+  return true;
+}
+
+export function hook([hook, qualifierObj = {}], attributes = {}) {
+  let qualifiers;
 
-    return returnWhenTesting(config, decorateHook(delimit(hook), qualifiers));
+  if (!isEmpty(qualifierObj) && !isEmpty(attributes)) {
+    throw new Error('Either provide your own qualifier object, or add attributes to the "hook" helper, not both.');
+  } else if (!isEmpty(qualifierObj)) {
```

Comment:
`throw` will stop the execution of the code so the code after shouldn't be executed.

no need to add the `else`, it will make the code a bit clearer (not a big fan of if/else branches)